### PR TITLE
feat: support team organization role assignment

### LINKED ISSUE #2314: [FEAT]: add support for pre-defined organization roles
### Describe the need

As announced https://github.blog/changelog/2024-07-10-pre-defined-organization-roles-that-grant-access-to-all-repositories/ Github has added support for pre-defined organisation roles.
To add new resource like `github_team_organization_role` similar to [`github_team_repository`](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_repository) to assign team to pre-defined role.
```
resource "github_team" "some_team" {
  name        = "SomeTeam"
  description = "Some cool team"
}

resource "github_team_organization_role" "some_team_role" {
  team_id    = github_team.some_team.slug
  role = "all_repo_read"
}
```
API:
- [Assign an organization role to a team](https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28#assign-an-organization-role-to-a-team)
- [Get all organization roles for an organization](https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28#get-all-organization-roles-for-an-organization)
- [Remove an organization role from a team](https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28#remove-an-organization-role-from-a-team)

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by felixlut**
Resolves #2314

----

### Before the change?

* It was not possible to assign an organization role to a team with the Terraform provider

### After the change?

* Implemented a `github_team_organization_role_assignment` resource which is used to assign an organization role to a team
* I've added the [`octokit/go-sdk`](https://github.com/octokit/go-sdk) since [``google/go-github``](https://github.com/google/go-github) lacked support for assigning a role to a team as far as I could see. go-github is still being used for this resource though, as go-sdk lacks support for pagination (https://github.com/microsoft/kiota/issues/4698)

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No, this is a brand new resource

----

### CODE REVIEW

File: github/config.go [hunk 1]
```diff
@@ -110,6 +112,31 @@ func (c *Config) NewRESTClient(client *http.Client) (*github.Client, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	if uv3.String() != "https://api.github.com/" {
+		uv3.Path = uv3.Path + "api/v3/"
+	}
+
+	octokitClient, err := pkg.NewApiClient(
+		// pkg.WithUserAgent("my-user-agent"), // Should this be set to terraform-provider-github or similar? Doesn't look like the user-agent is set for the other clients
```
Comment (felixlut) [inline github/config.go:127]: Seems like the provider is not setting the `user-agent` for the other clients, but maybe it should?

File: github/resource_github_team_organization_role_assignment.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		return err
+	}
+
+	roleID, err := strconv.ParseInt(roleIDString, 10, 32)
+	if err != nil {
+		return err
+	}
+
+	// There is no api for checking a specific team role assignment, so instead we iterate over all teams assigned to the role
+	// go-github pagination (https://github.com/google/go-github?tab=readme-ov-file#pagination)
```
Comment (felixlut) [inline github/resource_github_team_organization_role_assignment.go:126]: Using `google/go-github` here instead of `octokit/go-sdk` since the former doesn't support pagination

File: github/resource_github_team_organization_role_assignment_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccGithubTeamOrganizationRoleAssignment(t *testing.T) {
+
+	randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+
+	// Using the predefined roles since custom roles are a strictly Enterprise feature ((https://github.blog/changelog/2024-07-10-pre-defined-organization-roles-that-grant-access-to-all-repositories/))
```
Comment (felixlut) [inline github/resource_github_team_organization_role_assignment_test.go:15]: Since custom organization roles are a Enterprise level feature, I have elected to not use them in the tests and instead use the built-in roles instead. I've created a manual mapping for their `role_id`:s here. Not ideal, but I don't have a better way of testing this

**Comment by ChristopherMank** [comment]
Looking forward to this feature! Any thoughts on when it might be implemented?

**Comment by felixlut** [comment]
@ChristopherMank sometime after #2443 is merged probably. The bump of the SDK will include https://github.com/google/go-github/pull/3281, which makes it possible to use the [google/go-github](https://github.com/google/go-github) sdk entirely for this PR instead of the `[octokit/go-sdk](https://github.com/octokit/go-sdk)`. I want to avoid using it because of the reason mentioned [here](https://github.com/integrations/terraform-provider-github/pull/2316#issuecomment-2402699430)

**Comment by garettmd** [comment]
@felixlut any chance we could merge this? Looks like the #2443 was merged back in October, so hopefully there aren't any more blockers on this feature.

**Comment by felixlut** [comment]
@garettmd I'm not actively working on this. I think the latest update regarding Organization Roles support can be found in #2487. 

TLDR is that there are open PR:s for solving the issue, but we are lacking maintainer support for getting them over the line. The 2 latest updates to this repo (#2615, #2665) might offer some more context on that topic.

**Review by nickfloyd — APPROVED** [review]

**Comment by nickfloyd** [comment]
Fast follow: https://github.com/integrations/terraform-provider-github/pull/2787

**Comment by stevehipwell** [comment]
@nickfloyd this PR adds a duplicate resource to the `github_organization_role_team` resource added in #2487.
